src/Chainweb/Pact/SPV.hs

```diff
@@ -81,53 +89,78 @@ pactSPV cdbv l = SPVSupport (verifySPV cdbv l) (verifyCont cdbv l)
 verifySPV
     :: forall cas
     . MVar (CutDb cas)
+      -- ^ handle into the cut db
     -> Logger
+      -- ^ pact service logger
     -> Text
+      -- ^ TXOUT or TXIN - defines the type of proof
+      -- used in validation
     -> Object Name
+      -- ^ the 'TransactionOutputProof' object to validate
     -> IO (Either Text (Object Name))
 verifySPV cdbv l typ proof = readMVar cdbv >>= go typ proof
   where
     -- extract spv resources from pact object
     go s o cdb = case s of
-      "TXOUT" -> case txOutputProofOf o of
-        (Left !u) -> return $! Left u
-        Right t -> extractOutputs =<< verifyTransactionOutputProof cdb t
+      "TXOUT" -> case extractProof o of
+        Left !u -> return $! Left u
+        Right !t -> verifyTransactionOutputProof cdb t >>= extractOutputs handleResult
       "TXIN" -> return $ Left "TXIN is currently unsupported"
-      x -> return . Left
-        $ "TXIN or TXOUT must be specified to generate valid spv proofs: "
-        <> x
-
-    txOutputProofOf
-        :: Object Name
-        -> Either Text (TransactionOutputProof SHA512t_256)
-    txOutputProofOf o = k =<< toPactValue (TObject o def)
-      where
-        k = aeson (Left . pack) Right . fromJSON . toJSON
-
-    extractOutputs :: TransactionOutput -> IO (Either Text (Object Name))
-    extractOutputs (TransactionOutput t) =
-      case decodeStrict' t :: Maybe HashCommandResult of
-        Nothing -> internalError $
-          "unable to decode spv transaction output"
-        Just (CommandResult _ _ (PactResult (Right pv)) _ _ _ _) -> case fromPactValue pv of
-          (TObject !o _) -> return $! Right o
-          !o -> do
-            logLog l "ERROR" $ show o
-            return . Left $ pack "type error in associated pact transaction, should be object"
-        Just o -> do
-          logLog l "ERROR" $ show o
-          return . Left $ pack "Invalid command result in associated pact output"
+      t -> return . Left $!
+        "TXIN or TXOUT must be specified to generate valid spv proofs: " <> t
```

Comment:
we should probably ditch "TXIN" and just say `"unsupported SPV type: " <> t`

Reply:
Done